### PR TITLE
added query parameters, tests and open api

### PR DESCRIPTION
**Pull request by Schereo**

### CODE REVIEW

File: api/.prettierrc [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	"semi": true,
 	"trailingComma": "none",
 	"singleQuote": true,
-	"printWidth": 150,
+	"printWidth": 140,
```
Comment (Schereo) [inline api/.prettierrc:5]: for better line wrapping

File: api/src/routes/channel/index.ts [hunk 2]
```diff
@@ -34,14 +37,23 @@ export class ChannelRoutes {
...
 
 			const limitParam = parseInt(<string>req.query.limit, 10);
 			const indexParam = parseInt(<string>req.query.index, 10);
 			const limit = isNaN(limitParam) || limitParam == 0 ? undefined : limitParam;
 			const index = isNaN(indexParam) ? undefined : indexParam;
 			const ascending: boolean = <string>req.query.asc === 'true';
-			const options = limit !== undefined && index !== undefined ? { limit, index, ascending } : { ascending };
-
+			const options: ChannelLogRequestOptions = limit !== undefined && index !== undefined ? { limit, index, ascending } : { ascending };
+			options.startDate = startDate;
```
Comment (dominic22) [inline api/src/routes/channel/index.ts:55]: please set them in line 54

**Comment by dominic22** [comment]
please always link the github issue in the PRs

File: api/src/database/channel-data.ts [hunk 3]
```diff
@@ -10,12 +11,20 @@ const getIndex = (link: string, identityId: string) => `${link}-${identityId}`;
...
+	options: ChannelLogRequestOptions,
 	secret: string
 ): Promise<ChannelData[]> => {
-	const { ascending, index, limit } = options;
+	const { ascending, index, limit, startDate, endDate } = options;
 
-	const query = { channelAddress, identityId };
+	let created: { $gte?: string; $lte?: string };
+	if (startDate && endDate) created = { $gte: startDate, $lte: endDate };
+	else if (startDate && !endDate) created = { $gte: startDate };
```
Comment (dominic22) [inline api/src/database/channel-data.ts:21]: General suggestion: Please always use {} although it is a online if statement and we remove this code

File: api/src/database/channel-data.ts [hunk 4]
```diff
@@ -10,12 +11,20 @@ const getIndex = (link: string, identityId: string) => `${link}-${identityId}`;
...
 	identityId: string,
-	options: { limit?: number; index?: number; ascending: boolean },
+	options: ChannelLogRequestOptions,
 	secret: string
 ): Promise<ChannelData[]> => {
-	const { ascending, index, limit } = options;
+	const { ascending, index, limit, startDate, endDate } = options;
 
-	const query = { channelAddress, identityId };
+	let created: { $gte?: string; $lte?: string };
```
Comment (dominic22) [inline api/src/database/channel-data.ts:19]: This wants a refactoring. Please replace the code as following:

```
const startFilter = startDate ? {$gte:startDate}: {}
const endFilter = endDate ? {$lte: endDate}:{}

const query = {
		channelAddress,
		identityId,
                 ...startFilter,
                 ...endFilter
}
```

**Review by dominic22 — APPROVED** [review]

**Pull request merged**